circuit-overlay/src/commonMain/kotlin/com/slack/circuit/overlay/Overlay.kt

```diff
@@ -166,6 +171,30 @@ public interface Overlay<Result : Any> {
   @Composable public fun Content(navigator: OverlayNavigator<Result>)
 }
 
+/**
+ * An [Overlay] that can be animated in and more interestingly out _after_ it has returned a
+ * [Result]. The [Overlay] is animated in and out by its [enterTransition] and [exitTransition],
+ * [AnimatedContent] is executed with with [AnimatedVisibilityScope] so that child animations can be
+ * coordinated with the overlay's animations.
+ */
+public abstract class AnimatedOverlay<Result : Any>(
+  public val enterTransition: EnterTransition,
+  public val exitTransition: ExitTransition,
+) : Overlay<Result> {
+  @Composable
+  final override fun Content(navigator: OverlayNavigator<Result>) {
+    AnimatedContent(
+      targetState = Unit,
+      transitionSpec = { EnterTransition.None togetherWith ExitTransition.None },
```

Comment:
I believe we should be using `enterTransition` and `exitTransition` here

Reply:
Ahhh good catch #1090